_posts/articles/2021-02-14-setting-up-techdocs-on-backstage.md

```diff
@@ -0,0 +1,210 @@
+
+- The techdocs container which can be found on Docker-hub, It builds static content through MKDocs
+- The Techdocs backend plugin This is the backend part of the techdocs plugin.
+- The Techdocs CLI, Command line tool for managing techdocs sites in backstage
+- The Techdocs reader, it fetch's remotes pages , run transformers against them and renders them in a shadow DOM
+- Transformers API is a function that takes in parameters from the reader component and returns a function which gets passed the dom of the fetched page.
+
+Architecture
+
+![Architecture diagram](https://backstage.io/docs/assets/techdocs/architecture-basic.drawio.svg)
+
+When you open a techdocs sites a request is made the techdocs reader calls the techdocs-backend with the entity id and the path of the current page, the response contains the static content
+
+The static content contains HTML and CSS, javascript is removed for security reasons
+
+Transforms are then applied which modiy the generated static HTML files for a number of reasons like removing certain headers etc
+
+For the following instructions we will use the local file storage but it is better to use cloud storage like S3.
+
+## **Install Backstage**
+
+### Perquisites
+Mac running MacOS
+
+Node version 14.x
+
+cookiecutter
+
+If you already have backstage installed then skip this.
+
+```jsx
+npx @backstage/create-app
+```
+
+You are asked some questions on setup, The recommendation for this tutorial is to go with SQLLite.
+
+![npx install](/images/techdocs/Untitled.png)
+
+If successful you will see this message
+
+![successful](/images/techdocs/01.png)
+
+cd into your directory  and run 
+
+```jsx
+yarn workspace backend start
+```
+
+![yarn workspace](/images/techdocs/02.png)
+
+Open a new terminal window and 
+
+```jsx
+run yarn start
+```
+
+![yarn start](/images/techdocs/03.png)
+
+If successful a browser window will open up and you should be presented with a window
+
+![successfull register](/images/techdocs/04.png)
+
+Well done, you have successfully installed backstage on your machine.
+
+## Setup TechDocs
+
+Historically you had to manually add Techdocs , the  latest version of create-app bundles tech docs.
+
+to verify this you should be able to see entries for the following plugin 
+
+```jsx
+'@backstage/plugin-techdocs';
+  
+```
+
+in the following files
+
+```jsx
+packages/app/src/plugins.ts
+packages/app/src/App.tsx
+```
+
+## Creating and publishing techdocs
+
+To create docs manually from scratch click on create component
+
+![manuall](/images/techdocs/05.png)
+
+From here choose the Documentation template
+
+![template](/images/techdocs/06.png)
+
+Fill out Name and Description
+
+![Name Desc](/images/techdocs/07.png)
+
+Type in the owner, the Github repo you want to call it , ensure their is no Github repo that exists with the same name or the Templater will fail
+
+![owner](/images/techdocs/08.png)
+
+Once you click on create you will be presented with a Create component status popup.
```

Comment:
This will fail unless the user has a `GITHUB_TOKEN` set. They haven't been instructed to do this anywhere in the tutorial. I think this needs to be covered.

Once [this PR](https://github.com/RoadieHQ/marketing-site/pull/115) is merged we will have a dedicated tutorial available for generating the `GITHUB_TOKEN`. Perhaps we could link this in the post?